aws/resource_aws_internet_gateway_attachment.go

```diff
@@ -0,0 +1,182 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/aws-sdk-go-base/tfawserr"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	tfec2 "github.com/terraform-providers/terraform-provider-aws/aws/internal/service/ec2"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/service/ec2/finder"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/service/ec2/waiter"
+)
+
+func resourceAwsInternetGatewayAttachment() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsInternetGatewayAttachmentCreate,
+		Read:   resourceAwsInternetGatewayAttachmentRead,
+		Delete: resourceAwsInternetGatewayAttachmentDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"vpc_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"internet_gateway_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceAwsInternetGatewayAttachmentCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+
+	igwID := d.Get("internet_gateway_id").(string)
+	vpcID := d.Get("vpc_id").(string)
+
+	input := &ec2.AttachInternetGatewayInput{
+		InternetGatewayId: aws.String(igwID),
+		VpcId:             aws.String(vpcID),
+	}
+
+	log.Printf("[DEBUG] Creating internet gateway attachment")
+	err := resource.Retry(2*time.Minute, func() *resource.RetryError {
+		_, err := conn.AttachInternetGateway(input)
+		if err == nil {
+			return nil
+		}
+		if tfawserr.ErrCodeEquals(err, tfec2.ErrCodeInternetGatewayIDNotFound) {
+			return resource.RetryableError(err)
+		}
+
+		return resource.NonRetryableError(err)
+	})
+	if isResourceTimeoutError(err) {
+		_, err = conn.AttachInternetGateway(input)
+	}
+	if err != nil {
+		return fmt.Errorf("Error creating Internet Gateway attachment: %w", err)
+	}
+
+	d.SetId(fmt.Sprintf("%s:%s", vpcID, igwID))
```

Comment:
Add `InternetGatewayCreateID` in https://github.com/hashicorp/terraform-provider-aws/blob/master/aws/internal/service/ec2/id.go.

Reply:
done